src/server/service/kv.rs

```diff
@@ -458,29 +477,31 @@ impl<T: RaftStoreRouter<RocksEngine> + 'static, E: Engine, L: LockManager> Tikv
             cb,
         );
 
-        let future = AndThenWith::new(res, f.map_err(Error::from))
-            .and_then(|v| {
-                let mut resp = PhysicalScanLockResponse::default();
-                match v {
-                    Ok(locks) => resp.set_locks(locks.into()),
-                    Err(e) => resp.set_error(format!("{}", e)),
-                }
-                sink.success(resp).map_err(Error::from)
-            })
-            .map(move |_| {
-                GRPC_MSG_HISTOGRAM_STATIC
-                    .physical_scan_lock
-                    .observe(duration_to_sec(begin_instant.elapsed()))
-            })
-            .map_err(move |e| {
-                debug!("kv rpc failed";
-                    "request" => "physical_scan_lock",
-                    "err" => ?e
-                );
-                GRPC_MSG_FAIL_COUNTER.physical_scan_lock.inc();
-            });
+        let task = async move {
+            let res = match res {
+                Err(e) => Err(e),
+                Ok(_) => f.await?,
+            };
+            let mut resp = PhysicalScanLockResponse::default();
+            match res {
+                Ok(locks) => resp.set_locks(locks.into()),
+                Err(e) => resp.set_error(format!("{}", e)),
```

Comment:
Is it correct to skip extracting region error? /cc @youjiali1995

Reply:
Yes. It's a store-level command.